### PR TITLE
Add support of XDP driver

### PR DESCRIPTION
**Pull request by sergeymatov**
Switching on Ubuntu Focal Fossa 20.04 release
VPP based on 20-09 stable branch
Libbpf pumped to 0.2.0

E2E tests now have E2E_XDP env variable. It's needed to configure VPP interfaces in respect to chosen driver. XDP-based VPP interfaces will be used if E2E_XDP is not empty, af_packet otherwise

Example:
    make e2e-debug \
    E2E_QUICK=1 \
    E2E_XDP=y

Limitations: XDP designed to have single memory page for packet. Entire packet size should not exceed 4096 bytes, so MTU is limited to 3520 bytes

### CODE REVIEW

**Comment by RoadRunnr** [comment]
some of the build system changes seem to be related to the use of the old 18.04 Ubuntu LTS release. VPP 20.09 support the 20.04 LTS release and I think we should switch to that.

**Comment by sergeymatov** [comment]
@RoadRunnr @ivan4th 
Please take a look on this request again
It has several successful e2e test runs with both xdp/af_packet

**Review by ivan4th — CHANGES_REQUESTED** [review]

File: test/e2e/framework/framework.go [hunk 1]
```diff
@@ -210,9 +210,9 @@ func (f *Framework) AfterEach() {
 				os.Remove(filename)
 			}
 			logrus.WithField("testDir", targetDir).Info("test artifacts for the failed testcase")
-		} else if f.VPPCfg.BaseDir != "" {
+		} /*else if f.VPPCfg.BaseDir != "" {
 			ExpectNoError(os.RemoveAll(f.VPPCfg.BaseDir))
-		}
+		}*/
```
[the remark this answers is not in the document]
Comment (sergeymatov) [reply test/e2e/framework/framework.go]: Yes, fixed

File: test/e2e/framework/upgmode.go [hunk 2]
```diff
@@ -67,6 +67,7 @@ func pgwVPPConfigIPv4() vpp.VPPConfig {
 				// using L3 capture because of tun
 				// (no Ethernet headers)
 				L3Capture: true,
+				MTU: 1464,
```
[the remark this answers is not in the document]
Comment (sergeymatov) [reply test/e2e/framework/upgmode.go]: Added in a vpp constants set

File: test/e2e/upg_e2e.go [hunk 3]
```diff
@@ -597,11 +597,16 @@ func describeMTU(mode framework.UPGMode, ipMode framework.UPGIPMode) {
...
 		f := framework.NewDefaultFramework(mode, ipMode)
 		for i := range f.VPPCfg.Namespaces {
-			f.VPPCfg.Namespaces[i].MTU = 1500
+			if strings.HasPrefix(f.VPPCfg.Namespaces[i].Name, "grx") {
+				continue
+			}
+			f.VPPCfg.Namespaces[i].MTU = 1460
 		}
-		f.GTPUMTU = 9000
+		f.GTPUMTU = 1460
```
[the remark this answers is not in the document]
Comment (sergeymatov) [reply test/e2e/upg_e2e.go]: Constant fixed
I suppose  we have to check max XDP MTU, added a TODO for this

File: test/e2e/vpp/vpp.go [hunk 4]
```diff
@@ -38,7 +38,7 @@ const (
 	VPP_STARTUP_TIMEOUT     = 30 * time.Second
 	VPP_REPLY_TIMEOUT       = 5 * time.Second
 	NSENTER_CMD             = "nsenter"
-	DEFAULT_MTU             = 9000
+	DEFAULT_MTU             = 1500
```
[the remark this answers is not in the document]
Comment (sergeymatov) [reply test/e2e/vpp/vpp.go:41]: Constant added

File: test/e2e/upg_e2e.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (RoadRunnr) [reply test/e2e/upg_e2e.go]: > Constant fixed
> I suppose we have to check max XDP MTU, added a TODO for this

found that [her](https://people.netfilter.org/hawk/presentations/ebplane2019/xdp-bpf-building-blocks.pdf)
`XDP max MTU: 3520 bytes (page_size(4096) - headroom(256) - shinfo(320))`

**Comment by ivan4th** [comment]
LGTM. There are some quirks in the "MTU corner cases" e2e test case but I will try to fix it myself in a followup.

**Review by RoadRunnr — APPROVED** [review]

**Review by ivan4th — APPROVED** [review]
